2014-01-05  Magnus Granberg  <zorry@gentoo.org>

	We copy main.py from portage and patch it.
	Use or own patched actions.
	We pass build_dict to some functions.

--- a/zobcs/pym/main.py	2013-03-22 17:57:23.000000000 +0100
+++ b/zobcs/pym/main.py	2012-12-06 03:32:56.104889716 +0100
@@ -11,7 +11,7 @@ portage.proxy.lazyimport.lazyimport(glob
 	'logging',
 	'portage.util:writemsg_level',
 	'textwrap',
-	'_emerge.actions:load_emerge_config,run_action,' + \
+	'zobcs.actions:load_emerge_config,run_action,' + \
 		'validate_ebuild_environment',
 	'_emerge.help:help@emerge_help',
 )
@@ -968,15 +968,20 @@ def profile_check(trees, myaction):
 		return 1
 	return os.EX_OK
 
-def emerge_main(args=None):
+def emerge_main(args=None, build_dict=None):
 	"""
 	@param args: command arguments (default: sys.argv[1:])
 	@type args: list
+	@param build_dict: info of the build_job
+	@type build_dict: dict
 	"""
 	if args is None:
 		args = sys.argv[1:]
 
 	args = portage._decode_argv(args)
+	
+	if build_dict is None:
+		build_dict = {}
 
 	# Disable color until we're sure that it should be enabled (after
 	# EMERGE_DEFAULT_OPTS has been parsed).
@@ -1028,7 +1028,7 @@ def emerge_main(args=None):
 		parse_opts(tmpcmdline)
 
 	try:
-		return run_action(emerge_config)
+		return run_action(emerge_config, build_dict)
 	finally:
 		# Call destructors for our portdbapi instances.
 		for x in emerge_config.trees.values():
